wpimath/src/main/native/include/frc/trajectory/constraint/EllipticalRegionConstraint.h

```diff
@@ -16,6 +19,8 @@ namespace frc {
 /**
  * Enforces a particular constraint only within an elliptical region.
  */
+template <typename Constraint, typename = std::enable_if_t<std::is_base_of_v<
+                                   TrajectoryConstraint, Constraint>>>
```

Comment:
Since this class was changed to store a unique_ptr, couldn't it use unique_ptr<TrajectoryConstraint>?  Does TrajectoryConstraint not have a virtual destructor?  Then only the constructor would need to be templated, not the entire class.

Reply:
When creating a copy of the constraint in the copy constructor, the type of the derived class needs to be known; therefore, the entire class has to be templated.

Reply:
Why?  You can just template the constructor function and make the copy there into the generic unique_ptr.

Reply:
Ah, okay.  I guess the only solution to that would be to use shared_ptr to handle the copy case.

But in that case why is it unique_ptr at all?  Can't you just store by value since you're templating the entire class?

Reply:
That is a good point 😛.. I can just store it by value.